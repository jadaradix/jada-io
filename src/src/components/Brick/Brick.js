import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components'

const StyledBrick = styled.section`
  padding: 1.5rem;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", sans-serif;
  font-weight: 400;
  text-size-adjust: 100%;
  -webkit-font-smoothing: antialiased;
  letter-spacing: 0.1px;
  border-radius: 4px;
  box-shadow: 0px 4px 16px rgba(0, 0, 0, 0.1);
  h1 {
    font-size: 2rem;
    font-weight: 800;
    color: rgb(96, 96, 96);
    line-height: 3rem;
    // background-color: green;
  }
  p {
    color: rgb(48, 48, 48);
    line-height: 1.2rem;
    // background-color: red;
  }
  h1 + p {
    margin-top: 0.5rem;
  }
  p + p {
    margin-top: 1rem;
  }
`;

const Brick = ({ children, backgroundColor }) => (
  <StyledBrick style={{backgroundColor}}>
    {children}
  </StyledBrick>
);

Brick.propTypes = {
  backgroundColor: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};

Brick.defaultProps = {
  backgroundColor: 'rgb(250, 250, 250)'
};

export default Brick;
