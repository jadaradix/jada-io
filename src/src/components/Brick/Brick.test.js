import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import Brick from './Brick'

it('renders', () => {
  const tree = renderer
    .create(<Brick />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
