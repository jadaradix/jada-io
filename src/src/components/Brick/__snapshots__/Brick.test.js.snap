// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  padding: 1.5rem;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",sans-serif;
  font-weight: 400;
  -webkit-text-size-adjust: 100%;
  text-size-adjust: 100%;
  -webkit-font-smoothing: antialiased;
  -webkit-letter-spacing: 0.1px;
  -moz-letter-spacing: 0.1px;
  -ms-letter-spacing: 0.1px;
  letter-spacing: 0.1px;
  border-radius: 8px;
  box-shadow: 0 2px 8px 0 rgba(0,0,0,0.1);
}

.c0 h1 {
  font-size: 2rem;
  font-weight: 800;
  color: rgb(96,96,96);
  line-height: 3rem;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
}

.c0 p {
  color: rgb(48,48,48);
  line-height: 1.2rem;
}

.c0 h1 + p {
  margin-top: 0.5rem;
}

.c0 p + p {
  margin-top: 1rem;
}

.c0 p + a {
  margin-top: 1rem;
}

.c0 p + ul {
  margin-top: 1rem;
}

.c0 ul + p {
  margin-top: 1rem;
}

.c0 ul {
  margin-left: 2rem;
  color: rgb(48,48,48);
}

.c0 ul li {
  line-height: 1.2rem;
}

.c0 ul li:not(:last-of-type) {
  margin-bottom: 0.5rem;
}

.c0 a {
  color: rgb(255,45,85);
}

.c0 a + a {
  margin-top: 0.5rem;
}

<section
  className="c0"
  style={
    Object {
      "backgroundColor": "rgb(255, 255, 255)",
    }
  }
/>
`;
